### PR TITLE
Part 1 for refactoring bittensor/subtensor.py

### LINKED ISSUE #1908: bittensor.subtensor module refactoring
### Is your feature request related to a problem? Please describe.

As a part of https://github.com/opentensor/bittensor/issues/1882

### Describe the solution you'd like

- [x] Style refactoring
- [x] Сode refactoring

- [ ] ~~Implement the ability to use functions in an asynchronous implementation.~~ Decided to do this in different PR

### PR DESCRIPTION
**Pull request by basfroman**
Since the refactoring of this module would turn into a large project, it was decided to divide it into several stages.

The current part affects everything up to the subtensor.get_transfer_fee() method

### CODE REVIEW

File: bittensor/subtensor.py [hunk 1]
```diff
@@ -261,22 +262,31 @@ def __init__(
...
+            _logger.error(
+                f"Could not connect to %s network with %s chain endpoint. Exiting...",
+                self.network,
+                self.chain_endpoint,
             )
-            bittensor.logging.info(
-                f"You can check if you have connectivity by runing this command: nc -vz localhost {self.chain_endpoint.split(':')[2]}"
+            _logger.info(
+                "You can check if you have connectivity by running this command: nc -vz localhost %s",
+                self.chain_endpoint.split(":")[2],
```
Comment (thewhaleking) [inline bittensor/subtensor.py]: Why the change from f-strings to the older-style %s here?

File: bittensor/subtensor.py [hunk 2]
```diff
@@ -496,7 +534,7 @@ def nominate(
 
     def delegate(
         self,
-        wallet: "bittensor.wallet",
+        wallet: bittensor.wallet,
```
Comment (thewhaleking) [inline bittensor/subtensor.py:535]: This probably doesn't matter here, but don't we tend to use quotes around bittensor typing to avoid circular imports?

File: bittensor/utils/balance.py [hunk 3]
```diff
@@ -258,9 +258,9 @@ def from_float(amount: float):
...
 
     @staticmethod
-    def from_tao(amount: float):
+    def from_tao(amount: Union[int, float]):
         """
-        Given tao (float), return Balance object with rao(int) and tao(float), where rao = int(tao*pow(10,9))
+        Given tao (int | float), return Balance object with rao(int) and tao(float), where rao = int(tao*pow(10,9))
 
         Args:
             amount: The amount in tao.
```
Comment (thewhaleking) [inline bittensor/utils/balance.py]: While `from_tao` and `from_rao` accept ints or floats, I think the intention of keeping them `tao: float` and `rao: int` is a good idea.

**Review by thewhaleking — COMMENTED** [review]
See line comments. Not necessarily requesting changes, but want to discuss these couple items first.

File: bittensor/subtensor.py (hunk 2 above)
Comment (basfroman) [reply bittensor/subtensor.py:535]: As we decided in [RFC 003: Bittensor Coding Standards](https://docs.google.com/document/d/1ERaRsZY-SFVJY0a179_rX2-8Ff-aFQWTUNn-gXj3UiI/edit#heading=h.cec1zrvrfmzv) -> Approach -> Typing :: `Prefer imported types for annotations; if not possible, use type names in quotes (for example to avoid circular imports).`
In this case there are no circular imports.

File: bittensor/utils/balance.py (hunk 3 above)
Comment (basfroman) [reply bittensor/utils/balance.py]: In general, I added this because in different parts of the code for both methods we pass both  types (int and float).
Let's return it as it was and more strictly control which types we will pass.

File: bittensor/subtensor.py (hunk 1 above)
Comment (basfroman) [reply bittensor/subtensor.py]: Switch back

**Review by thewhaleking — APPROVED** [review]

**Review by garrett-opentensor — APPROVED** [review]
